otlp/common.go

```diff
@@ -43,6 +43,9 @@ const (
 // The limit is enforced in retriever (in private honeycomb code), in varstring.go.
 const fieldSizeMax = math.MaxUint16
 
+// max depth for flattening nested attributes
+const maxDepth = 10
```

Comment:
This feels very deep to me. My off-the-cuff would have been 5.

Reply:
We only support 5 levels for the events endpoint:
<img width="376" alt="image" src="https://github.com/honeycombio/husky/assets/6309070/c67a77bd-c682-43bc-8b53-637819c52818">

My take is -- let's match there. We can always go deeper if we need to, but it might be harder to back out if we need to.

Reply:
Updated to 5